### ISSUE TITLE
NullPointerException of TwinTags When DeviceTwinSample Run

### ISSUE THREAD

**Issue by Hugo-X** [Azure/azure-iot-sdk-java#71]
- **OS and version used:**   Windows 7  

- **Java runtime used:**  Java(TM) SE Runtime Environment (build 1.8.0_121-b13)  

- **SDK version used:**  iot-service-client 1.2.17 

- **Source version release 2017-3-28**

# Description of the issue:

Download latest code of the master branch
Compile and run DeviceTwinSample
NullPointerException was thrown by TwinTags.update method if  the device has **Tags value**
After clear the Tags Value of device run DeviceTwinSample again, the process finished without error.

# Code sample exhibiting the issue:

Only modification about connection string  

# Console log of the issue:

**With Tags Value**
```
Starting sample...
Getting device twin
Exception in thread "main" java.lang.NullPointerException
	at com.microsoft.azure.sdk.iot.deps.serializer.TwinTags.update(TwinTags.java:145)
	at com.microsoft.azure.sdk.iot.deps.serializer.Twin.updateTwin(Twin.java:674)
	at com.microsoft.azure.sdk.iot.service.devicetwin.DeviceTwin.getTwin(DeviceTwin.java:132)
	at samples.com.microsoft.azure.sdk.iot.DeviceTwinSample.main(DeviceTwinSample.java:39)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

**Without Tags Value**
```
Starting sample...
Getting device twin
Device ID: xxtest
Tags:{}
Desired Properties: {k1=v1}

Shutting down sample...

Process finished with exit code 0
```

**Comment by prmathur-microsoft** [comment]
how are you updating tags ? by default java sdk does not yet let you update tags. Are you using any other SDK ?

**Comment by Hugo-X** [comment]
Tags were updated by DeviceExplorer.  

https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2017-3-23

**Comment by prmathur-microsoft** [comment]
Thank you. I'am able to reproduce. I'll push in a fix soon.

**Comment by prmathur-microsoft** [comment]
Fix checked in @0a16c5e182bd1389247e4b28c28faa35c6025de2.
Please re-open if you see issues.
